pkg/controller/controller.go

```diff
@@ -702,8 +805,52 @@ func (p *csiProvisioner) volumeHandleToId(handle string) string {
 	return handle
 }
 
-// getSecretReference returns a reference to the secret specified in the given nameKey and namespaceKey parameters, or an error if the parameters are not specified correctly.
-// if neither the name or namespace parameter are set, a nil reference and no error is returned.
+func getSecretNameAndNamespaceTemplate(secret secretParamsType, storageClassParams map[string]string) (nameTemplate, namespaceTemplate string, err error) {
```

Comment:
maybe add comments about this function? It verify the parameters and then get them. How about rename to "verifyAndGet..." since one major part of this is verifying?